ts/WoltLabSuite/Core/ColorUtil.ts

```diff
@@ -158,40 +158,162 @@ export function hexToRgb(hex: string): RGB | typeof Number.NaN {
   return Number.NaN;
 }
 
+/**
+ * @since 5.5
+ */
+function rgbComponentToHex(component: number): string {
+  if (component < 0 || component > 255) {
+    throw new Error(`Invalid RGB component value '${component}' given.`);
+  }
+
+  return component.toString(16).padStart(2, "0").toUpperCase();
+}
+
 /**
  * Converts a RGB into HEX.
- *
- * @see  http://www.linuxtopia.org/online_books/javascript_guides/javascript_faq/rgbtohex.htm
  */
-export function rgbToHex(r: number, g: number, b: number): string {
-  const charList = "0123456789ABCDEF";
+export function rgbToHex(r: string): string;
+export function rgbToHex(r: number, g: number, b: number): string;
+export function rgbToHex(r: string | number, g?: number, b?: number): string {
+  if (g === undefined) {
+    const match = /^rgba?\((\d+), ?(\d+), ?(\d+)(?:, ?[0-9.]+)?\)$/.exec(r.toString());
+    if (match) {
+      r = +match[1];
+      g = +match[2];
+      b = +match[3];
+    } else {
+      throw new Error("Invalid RGB data given.");
+    }
+  }
+
+  return rgbComponentToHex(r as number) + rgbComponentToHex(g) + rgbComponentToHex(b!);
+}
+
+/**
+ * @since 5.5
+ */
+function alphaToHex(alpha: number): string {
+  if (alpha < 0 || alpha > 1) {
+    throw new Error(`Invalid alpha value '${alpha}' given.`);
+  }
+
+  return Math.round(alpha * 255)
+    .toString(16)
+    .padStart(2, "0")
+    .toUpperCase();
+}
 
+/**
+ * Converts a RGBA value into a HEX value.
+ *
+ * @since 5.5
+ */
+export function rgbaToHex(rgba: RGBA): string;
+export function rgbaToHex(r: number, g: number, b: number, a: number);
+export function rgbaToHex(r: RGBA | number, g?: number, b?: number, a?: number): string {
   if (g === undefined) {
-    if (/^rgba?\((\d+), ?(\d+), ?(\d+)(?:, ?[0-9.]+)?\)$/.exec(r.toString())) {
-      r = +RegExp.$1;
-      g = +RegExp.$2;
-      b = +RegExp.$3;
+    const rgba = r as RGBA;
+    return rgbToHex(rgba.r, rgba.g, rgba.b) + alphaToHex(rgba.a);
+  }
+
+  return rgbToHex(r as number, g, b!) + alphaToHex(a!);
+}
+
+/**
+ * Returns the textual representation of a RGBA value.
+ *
+ * @since 5.5
+ */
+export function rgbaToString(rgba: RGBA): string {
+  return `rgba(${rgba.r}, ${rgba.g}, ${rgba.b}, ${rgba.a})`;
+}
+
+/**
+ * @since 5.5
+ */
+function getColorChecker(): HTMLSpanElement {
+  let colorChecker = document.getElementById("jsColorUtilColorChecker");
+  if (colorChecker === null) {
+    colorChecker = document.createElement("span");
+    colorChecker.id = "jsColorUtilColorChecker";
+    document.body.appendChild(colorChecker);
```

Comment:
Do you really need to inject the element into the DOM?

Reply:
Yes, for `window.getComputedStyle()`.